### PR TITLE
Runtimebh

### PR DESCRIPTION
**Pull request by sbird**
Remove the BLACK_HOLES runtime option. Overall cost is one float in sph_particle_data.

The BlackHoleOn parameter was a little strange before: if BLACK_HOLES was compiled but BlackHoleOn = 0 then existing type 5 particles would in places be treated as black holes. This would never really happen but I fixed it anyway, as there might be a performance hit for non black hole runs otherwise.

### CODE REVIEW

**Comment by sbird** [comment]
Tested that this makes no difference to the output with the hydro example. It marginally speeds up the no-black-hole case if -DBLACK_HOLES is defined, and has a 3% performance hit compared to the case where it is not defined.

**Comment by sbird** [comment]
Once this series of PRs removing the last of the compile-time options is in, I think I will look into shrinking sph_particle_data!

**Comment by sbird** [comment]
I spoke too soon: it has a 3% performance hit over the first few timesteps and a 1% performance *increase* over the first 30!

**Comment by sbird** [comment]
I think the performance changes I am measuring are really just random noise caused by running on whatever firefox is doing at the time. This PR passes tests and I think can be merged straightforwardly after review.

File: libgadget/blackhole.c [hunk 1]
```diff
@@ -224,6 +224,20 @@ blackhole(ForceTree * tree, double * TimeNextSeedingCheck)
     N_sph_swallowed = N_BH_swallowed = 0;
 
     /* Let's determine which particles may be swalled and calculate total feedback weights */
+    SPH_SwallowID = mymalloc("SPH_SwallowID", SlotsManager->info[0].size * sizeof(MyIDType));
```
Comment (rainwoodman) [inline libgadget/blackhole.c]: Is this a bug fix? Seems to be related to Yueying's Ptype=5 Mass not tracking BlackholeMass issue.

Comment (sbird) [reply libgadget/blackhole.c]: I don't think it is a bug fix - it is just allocating this memory within blackhole.c instead of in sph_particle_data

Comment (sbird) [reply libgadget/blackhole.c]: It might be that there is a bug there, but this does not change it as far as I can see.

**Pull request merged**